===== Patch instructions generated by SAVER =====
[+] { Insert tmp := configure_ciphers:cipher->cipher at 103 (line 150, column 7) and replace the original expression at 163 (line 204, column 4) under true }
===== Actual patch in diff format =====
@@ -196,13 +196,14 @@ configure_ciphers (grub_disk_t disk, const char *check_uuid,
     }
   else if (grub_memcmp (ciphermode, "lrw-", sizeof ("lrw-") - 1) == 0)
     {
+      tmp = cipher->cipher;
       mode = GRUB_CRYPTODISK_MODE_LRW;
       cipheriv = ciphermode + sizeof ("lrw-") - 1;
       if (cipher->cipher->blocksize != GRUB_CRYPTODISK_GF_BYTES)
 	{
 	  grub_crypto_cipher_close (cipher);
 	  grub_error (GRUB_ERR_BAD_ARGUMENT, "Unsupported LRW block size: %d",
-		      cipher->cipher->blocksize);
+		      tmp->blocksize);
 	  return NULL;
 	}
     }
